Tavisca.Bootcamp.LanguageBasics.Exercise2/Program.cs

```diff
@@ -13,18 +13,129 @@ static void Main(string[] args)
             Console.ReadKey(true);
         }
 
-        private static void Test(string[] postTimes, string[] showTimes, string expected)
-        {
+        private static void Test(string[] postTimes, string[] showTimes, string expected)
+        {
             var result = GetCurrentTime(postTimes, showTimes).Equals(expected) ? "PASS" : "FAIL";
             var postTimesCsv = string.Join(", ", postTimes);
             var showTimesCsv = string.Join(", ", showTimes);
-            Console.WriteLine($"[{postTimesCsv}], [{showTimesCsv}] => {result}");
+            Console.WriteLine($"[{postTimesCsv}], [{showTimesCsv}] => {result}");
+        }
+
+        public static string GetCurrentTime(string[] exactPostTime, string[] showPostTime)
+        {
+            DateTime previousStart = DateTime.Now;  // initialized to avoid compile-time error
+            DateTime previousEnd = DateTime.Now;  // initialized to avoid compile-time error
+            System.Console.WriteLine("I am Ironman");
+
+            for (int i = 0; i < exactPostTime.Length; i++) {
+                
+                // Get the first time interval
+                if (i == 0) {
+                    GetPossibleTimeInterval(exactPostTime[i], showPostTime[i], out previousStart, out previousEnd);
+                }
+                
+                else {
+                    DateTime currentStart;
+                    DateTime currentEnd;
+                    
+                    // Find time interval for current time 
+                    GetPossibleTimeInterval(exactPostTime[i], showPostTime[i], out currentStart, out currentEnd);
+
+                    #region  Find the intersection time interval between current and previous time stamps
+                        // p -> previous, c -> current, S -> start, E -> end
+
+                        // Timeline: [pS |cS pE| cE]
+                        if (currentStart >= previousStart && currentStart <= previousEnd && currentEnd >= previousEnd) {
+                                previousStart = currentStart;
+                                // previousEnd = previousEnd;  // Statement unnecessary
+                        }
+                        // Timeline: [cS |pS pE| cE]
+                        else if (currentStart <= previousStart && previousEnd <= currentEnd) {
+                                // previousStart = previousStart;  // Statement unnecessary

```

Comment:
Can you please remove all unwanted statements? It's just making the code more complex.